src/modules/landing_target_estimator/LandingTargetEstimator.cpp

```diff
@@ -100,67 +100,32 @@ void LandingTargetEstimator::update()
 		}
 	}
 
-	if (!_new_irlockReport) {
+	if (!_new_sensorReport) {
 		// nothing to do
 		return;
 	}
 
 	// mark this sensor measurement as consumed
-	_new_irlockReport = false;
+	_new_sensorReport = false;
 
-	if (!_vehicleAttitude_valid || !_vehicleLocalPosition_valid || !_vehicleLocalPosition.dist_bottom_valid) {
-		// don't have the data needed for an update
-		return;
-	}
-
-	if (!PX4_ISFINITE(_irlockReport.pos_y) || !PX4_ISFINITE(_irlockReport.pos_x)) {
-		return;
-	}
-
-	matrix::Vector<float, 3> sensor_ray; // ray pointing towards target in body frame
-	sensor_ray(0) = _irlockReport.pos_x * _params.scale_x; // forward
-	sensor_ray(1) = _irlockReport.pos_y * _params.scale_y; // right
-	sensor_ray(2) = 1.0f;
-
-	// rotate unit ray according to sensor orientation
-	_S_att = get_rot_matrix(_params.sensor_yaw);
-	sensor_ray = _S_att * sensor_ray;
-
-	// rotate the unit ray into the navigation frame
-	matrix::Quaternion<float> q_att(&_vehicleAttitude.q[0]);
-	_R_att = matrix::Dcm<float>(q_att);
-	sensor_ray = _R_att * sensor_ray;
-
-	if (fabsf(sensor_ray(2)) < 1e-6f) {
-		// z component of measurement unsafe, don't use this measurement
-		return;
-	}
-
-	float dist = _vehicleLocalPosition.dist_bottom - _params.offset_z;
-
-	// scale the ray s.t. the z component has length of dist
-	_rel_pos(0) = sensor_ray(0) / sensor_ray(2) * dist;
-	_rel_pos(1) = sensor_ray(1) / sensor_ray(2) * dist;
-
-	// Adjust relative position according to sensor offset
-	_rel_pos(0) += _params.offset_x;
-	_rel_pos(1) += _params.offset_y;
 
 	if (!_estimator_initialized) {
-		PX4_INFO("Init");
 		float vx_init = _vehicleLocalPosition.v_xy_valid ? -_vehicleLocalPosition.vx : 0.f;
 		float vy_init = _vehicleLocalPosition.v_xy_valid ? -_vehicleLocalPosition.vy : 0.f;
-		_kalman_filter_x.init(_rel_pos(0), vx_init, _params.pos_unc_init, _params.vel_unc_init);
-		_kalman_filter_y.init(_rel_pos(1), vy_init, _params.pos_unc_init, _params.vel_unc_init);
+		PX4_INFO("LTE: Init %.2f %.2f", (double)vx_init, (double)vy_init);
+		_kalman_filter_x.init(_sensor_report.rel_pos_x, vx_init, _params.pos_unc_init, _params.vel_unc_init);
+		_kalman_filter_y.init(_sensor_report.rel_pos_y, vy_init, _params.pos_unc_init, _params.vel_unc_init);
 
 		_estimator_initialized = true;
 		_last_update = hrt_absolute_time();
 		_last_predict = _last_update;
 
 	} else {
 		// update
-		bool update_x = _kalman_filter_x.update(_rel_pos(0), _params.meas_unc * dist * dist);
-		bool update_y = _kalman_filter_y.update(_rel_pos(1), _params.meas_unc * dist * dist);
+		bool update_x = _kalman_filter_x.update(_sensor_report.rel_pos_x,
```

Comment:
We could actually keep the name `dist` instead of `uncertainty_scale`, and then add a new line here that does
```
const float measurement_uncertainty = dist*dist

// and then
bool update_x = _kalman_filter_x.update(_sensor_report.rel_pos_x,
						_params.meas_unc * measurement_uncertainty);
bool update_y = _kalman_filter_y.update(_sensor_report.rel_pos_y,
						_params.meas_unc * measurement_uncertainty);
```
which would also avoid doing the same calculation twice.

Keeping the `dist` name would also help with this line further below, which is not intuitive:
```
_sensor_report.rel_pos_z = _uncertainty_scale;
```

Reply:
I agree.
I removed this uncertainty/dist from the UWB part and renamed it _dist_z .

Also:
```
		const float measurement_uncertainty = _params.meas_unc * _dist_z * _dist_z;
		bool update_x = _kalman_filter_x.update(_sensor_report.rel_pos_x, measurement_uncertainty);
		bool update_y = _kalman_filter_y.update(_sensor_report.rel_pos_y, measurement_uncertainty);
```
by moving the _params.meas_unc into the new constant we could save one more calculation.

Reply:
This will also resolve this:

`_uncertainty_scale = _vehicleLocalPosition.dist_bottom - _params.offset_z;`
> This is another thing that breaks if there are irlock_report and uwb_distance messages coming in.